dfastmi/batch/ConfigurationChecker.py

```diff
@@ -37,393 +37,114 @@
-    if cond not in config.sections():
-        return False
-    
-    if "Discharge" in config[cond]:
-        # if discharge is specified, it must be a number
-        if not is_float_str(config[cond]["Discharge"]):
+        except ValueError:
             return False
-    
-    if "Reference" not in config[cond]:
-        return False
-    
-    if "WithMeasure" not in config[cond]:
-        return False
-                
-    return True
-
-def is_float_str(string:str) -> bool:
-    """
-    Check if a string represents a (floating point) number.
-
-    Arguments
-    ---------
-    string : str
-        The string to be checked.
-
-    Returns
-    -------
-    success : bool
-        Boolean indicating whether the the string can be converted to a number or not.
-    """
-    try:
-        float(string)
-        return True
-    except ValueError:
-        return False
-
-def check_configuration_v2(rivers: RiversObject, config: configparser.ConfigParser) -> bool:
-    """
-    Check if a version 2 analysis configuration is valid.
-
-    Arguments
-    ---------
-    rivers: RiversObject
-        A dictionary containing the river data.
-    config : configparser.ConfigParser
-        Configuration for the D-FAST Morphological Impact analysis.
 
-    Returns
-    -------
-    success : bool
-        Boolean indicating whether the D-FAST MI analysis configuration is valid.
-    """
-    try:
-        reach = get_reach(rivers, config)
-    except:
-        return False
-
-    hydro_q = reach.hydro_q
-    n_cond = len(hydro_q)
-    
-    found = [False]*n_cond
-    
-    for i in range(n_cond):
-        success, found[i] = check_configuration_v2_cond(config, hydro_q[i])
-        if not success:
-            return False
-    
-    if not all(found):
-        return False
+        hydro_q = reach.hydro_q
+        q_stagnant = reach.qstagnant
+        if config.has_section("General") and config.has_option("General", "Qthreshold") :
+            try:
+                q_threshold = config.getfloat("General", "Qthreshold")
+            except ValueError:
+                q_threshold_str = config.get("General", "Qthreshold", fallback="")
+                ApplicationSettingsHelper.log_text(f"Please this configuration has in the General section an option for Qthreshold"
+                                                   f"but is not float but : {q_threshold_str}!" 
+                                                   f"Using q_stagnant as q_threshold : {q_stagnant}")
+                q_threshold = q_stagnant
+        else:
+            q_threshold = q_stagnant        
+        n_cond = len(hydro_q)
+
+        return_value = False
+        for i in range(n_cond):
+            if hydro_q[i] > q_threshold : 
+                success = self._check_configuration_cond(config, hydro_q[i])
+                if success and not return_value:
+                    return_value = True
+        return return_value
+
+    def _check_configuration_cond(self, 
```

Comment:
This method has the following code smell:

Refactor this function to reduce its Cognitive Complexity from 18 to the 15 allowed.

Reply:
unclear how to see if this is improved now, sonar cloud is not reporting it any more

Reply:
I think you fixed in in one of you latest commits.